lib/utils.tsx

```diff
@@ -104,19 +104,37 @@ const utils = {
     return result
   },
   toastError: (
-    err?: string | Error | { data: string } | unknown,
+    err?:
+      | string
+      | Error
+      | { data: { error?: string } | { message?: string } | unknown }
+      | unknown,
     showDialog?: boolean
   ) => {
     let message: string
 
     if (typeof err === 'string') {
       message = err
-    } else if (err instanceof Error) {
+    } else if (
+      err &&
+      typeof err === 'object' &&
+      'message' in err &&
+      typeof err.message === 'string'
+    ) {
       message = err.message
     } else if (err && typeof err === 'object' && 'data' in err) {
-      const errorWithData = err as { data: unknown }
+      const errorWithData = err as { data: { error?: string } | unknown }
       if (typeof errorWithData.data === 'string') {
         message = errorWithData.data
+      } else if (
+        errorWithData.data &&
+        typeof errorWithData.data === 'object' &&
+        'error' in errorWithData.data
+      ) {
+        message =
+          typeof errorWithData.data.error === 'string'
+            ? errorWithData.data.error
+            : 'Something went wrong'
```

Comment:
Ww can simplify it using optional chaining, WDYT? 

` const message = 
    typeof err === 'string' ? err :
    err instanceof Error ? err.message :
    typeof err === 'object' && err?.data?.error ? err.data.error :
    'Something went wrong';`

Reply:
I think in this case it's more readable with ifs

Reply:
If you prefer if statements, let's simplify the cases ? 

`let message: string;

if (typeof err === 'string') {
  message = err;
} else if (err instanceof Error) {
  message = err.message;
} else if (typeof err === 'object' && err?.data?.error) {
  message = err.data.error;
} else {
  message = 'Something went wrong';
}`

Reply:
```
else if (err instanceof Error) {
message = err.message;
}
```
⬆️ there are cases when `err` object is not `instanceof Error` and still can have the `message` property, therefore I changed this condition

```
else if (typeof err === 'object' && err?.data?.error) {
message = err.data.error;
}
```

⬆️ we also need to check that `err.data.error` is a string before assigning it to message

Reply:
ok , so we can try something like:

```
let message: string;

if (typeof err === 'string') {
  message = err;
} else if (err && typeof err === 'object' && 'message' in err && typeof err.message === 'string') {
  message = err.message;
} else if (err && typeof err === 'object' && err.data && typeof err.data === 'object' && typeof err.data.error === 'string') {
  message = err.data.error;
} else {
  message = 'Something went wrong';
}
```

Reply:
this way we're not covering this case:
```
if (typeof err.data === 'string') {
        message = err.data // errorWithData.data in the code
      }
```

Reply:
ok, so let's try that one:

```
let message: string;

if (typeof err === 'string') {
  message = err;
} else if (err && typeof err === 'object' && 'message' in err && typeof err.message === 'string') {
  message = err.message;
} else if (err && typeof err === 'object' && err.data) {
  if (typeof err.data === 'string') {
    message = err.data;
  } else if (typeof err.data === 'object' && 'error' in err.data && typeof err.data.error === 'string') {
    message = err.data.error;
  } else {
    message = 'Something went wrong';
  }
} else {
  message = 'Something went wrong';
}
```